app/src/androidTest/java/org/mozilla/fenix/perf/HomeFragmentBenchmark.kt

```diff
@@ -0,0 +1,129 @@
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.helpers.TestHelper
+import org.mozilla.fenix.ui.robots.mDevice
+import org.mozilla.fenix.utils.view.ViewHolder
+
+@RunWith(AndroidJUnit4::class)
+class HomeFragmentBenchmark {
+
+    private val resourceId = "sessionControlRecyclerView"
+
+    private val homeButton = "Home screen"
+    private lateinit var mockWebServer: MockWebServer
+
+    private lateinit var genericURL: TestAssetHelper.TestAsset
+
+    private lateinit var intent: Intent
+
+    @get:Rule
+    val benchmarkRule = BenchmarkRule()
+
+    @get:Rule
+    val homeActivityRule = HomeActivityTestRule(
+        skipOnboarding = true, skipShouldShowJumpBackIn = true
+    )
+
+    @get: Rule
+    val intentReceiverActivityTestRule = ActivityTestRule(
+        IntentReceiverActivity::class.java, true, false
+    )
+
+    @Before
+    fun setup() {
+        mockWebServer = MockWebServer().apply {
+            dispatcher = AndroidAssetDispatcher()
+            start()
+        }
+
+        genericURL = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        intent = Intent(Intent.ACTION_VIEW, genericURL.url)
+        intentReceiverActivityTestRule.launchActivity(intent)
+    }
+
+    /**
+     * Benchmarks the time taken from clicking the "Home Button" on the URL bar located on the
+     * BrowserFragment up until the HomeFragment is shown and usable.
+     *
+     * NOTE: According to Google's documentations, this test should be annotated with @UiThreadTest.
+     * However, doing so will make UiAutomator or Espresso API's unusable since the test will hang.
+     * Instead, we use the `runOnUiThread{}` which is the method that @UiThreadTest uses.
+     */
+    @Test
+    fun homeScreenBenchmark() {
+        var homeButton: View? = null
+
+        benchmarkRule.measureRepeated {
+            runWithTimingDisabled {
+                homeButton = getHomeButton(homeActivityRule.activity.findViewById<BrowserToolbar>(R.id.toolbar))
+            }
+            homeActivityRule.runOnUiThread {
+                homeButton!!.callOnClick()
+            }
+
+            // The homescreen could potentially take a while to show on the screen.
+            // Therefore, we should wait and make sure that the Main Thread is idled.
+            // `waitForIdleSync()` waits for the message queue to be empty. Source:
+            // https://android.googlesource.com/platform/frameworks/base/+/8f6f1f43eeb0d7263b626978ee2c21d4053bf610/core/java/android/app/Instrumentation.java#328
+            InstrumentationRegistry.getInstrumentation().waitForIdleSync()
+
+            runWithTimingDisabled {
+                // Make sure that everything is shown on screen. The recyclerview should be the last
+                // thing to be drawn.
+                assertNotNull(mDevice.findObject(By.res(TestHelper.packageName, resourceId)))
```

Comment:
nitpick (non-block): could we make the name of `resourceId` more explicit? Which resource ID is it?